### ISSUE TITLE
fix(model): adding meaningful fields for neural network builds

### ISSUE THREAD

**Issue by niyasrad** [qreater/Profiteer-IO#1]
# Description

Currently, the synthetic data used for model training, with around `720` Records only brings out `0.94` Accuracy Score, and it is in XGBoost, and with tweaks, is in Linear Regression. While the score seems above average, the results for the purchase prediction were totally off, partly due the amount of noise with the data. 

- Populate the synthetic data with meaningful fields, such as
    - `base_price`: Will be used by the model to compute price ratio (Discounts/ Premium Prices)
    - `time_of_day_bucket`: Corresponds to `Overnight`/`Morning`/`Afternoon`/`Evening`
    - `popularity_factor`: Internally computed by the device popularity, category popularity, and mostly `rating`.
